sale_prodlot_mrp/stock.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#  licence AGPL version 3 or later
+#  see licence in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2014 Akretion (http://www.akretion.com).
+#  @author David BEAL <david.beal@akretion.com>
+#
+##############################################################################
+
+from openerp.osv import orm
+
+
+class StockMove(orm.Model):
+    _inherit = 'stock.move'
+
+    #Complicated code that explode the prodlot number
+    #base function are really ugly, hard to inherit
+    def _action_explode(self, cr, uid, move, context=None):
+        """ only acts on sold products with phantom bom
+        """
+        if context is None:
+            context = {}
+        ctx = context.copy()
+        ctx['lot_base_name'] = move.restrict_lot_id.name
+        ctx['lot_index'] = 1
+        ctx['explode_lot'] = True
+
+        return super(StockMove, self)._action_explode(
+            cr, uid, move, context=ctx)
+
+    def _prepare_explode_move(self, cr, uid, move, line, context=None):
+        product_obj = self.pool['product.product']
+        lot_obj = self.pool['stock.production.lot']
+        if context is None:
+            context = {}
+        res = super(StockMove, self).\
+            _prepare_explode_move(cr, uid, move, line, context=context)
+        if context.get('explode_lot'):
+            if move.product_id.track_from_sale:
```

Comment:
Be carefull if you rename the flag into "sale_prodlot_generation" 
https://github.com/OCA/sale-workflow/pull/179/files#r33909242